### PR TITLE
Makefile: Separate scope of "checkstyle" and "test-checkstyle" for clarity

### PR DESCRIPTION
**Pull request by okurz**
We have "style checks" or static tests that are called by "prove", i.e.
t/00-tidy.t and t/01-compile-check-all.t, and we have independant tools,
i.e. shellcheck, yamllint and also perlcritic. These had been called in
"checkstyle" and tests called by "prove" have either been called
implicitly when calling all tests by prove without test selection
arguments or explicitly selected, e.g. as done by our circle CI tests.
As we moved the circle CI test selections into named Makefile targets we
ended up with "test-checkstyle" that sounds too similar to "checkstyle"
so that the scope for each was not clear.

This commit tries to improve that situation by putting all static tests
that are not called into a sub-target "test-checkstyle-standalone",
moves the call to tidy/compile with prove selection arguments into an
explicit target and combining all on the level of the existing
test group targets, e.g. "test-checkstyle". Individual targets are used
rather than having multiple shell lines within a single target to allow
easier independant calls as well as allowing make to run these tests in
parallel.

Also remove redundant CHECKSTYLE=1 on test-checkstyle call.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -103,24 +103,19 @@ install:
...
 
-.PHONY: checkstyle
-checkstyle: test-shellcheck test-yaml
-	PERL5LIB=lib/perlcritic:$$PERL5LIB perlcritic lib
-
 .PHONY: test
 ifeq ($(TRAVIS),true)
 test: run-tests-within-container
 else
 ifeq ($(CHECKSTYLE),0)
```
Comment (okurz) [inline Makefile]: @Martchus if I remember correctly you thought that needing two flags to unselect style checks and selecting individual tests could be improved. E.g. currently we need:

```sh
make test CHECKSTYLE=0 PROVE_ARGS=t/42-my-test.t
```

to run just a single test. Of course one can just use

```sh
make test-with-database PROVE_ARGS=t/42-my-test.t
```

instead but it's not less typing :)

We could do something like

```sh
make test TESTS=t/42-my-test.t
```

which internally switches off CHECKSTYLE. Would this be good? Any further ideas?

**Comment by okurz** [comment]
@kalikiana @perlpunk

File: Makefile (hunk 1 above)
Comment (perlpunk) [reply Makefile]: It's less typing for those who use tab completion ;-)

Comment (kalikiana) [reply Makefile]: I like `TESTS` in the sense that `PROVE_ARGS` takes more arguments than just the name of the test. You use it specify the *harness* for instance.

Comment (okurz) [reply Makefile]: Covered this in https://github.com/os-autoinst/openQA/pull/2847 . You can now approve this PR here :)

**Pull request merged**